presto-main/src/main/java/com/facebook/presto/operator/aggregation/InternalAggregationFunction.java

```diff
@@ -25,19 +26,39 @@
 import java.util.Optional;
 
 import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.Iterables.getOnlyElement;
 import static java.util.Objects.requireNonNull;
 
 public final class InternalAggregationFunction
 {
     private final String name;
     private final List<Type> parameterTypes;
-    private final Type intermediateType;
+    private final List<Type> intermediateType;
     private final Type finalType;
     private final boolean decomposable;
     private final boolean orderSensitive;
     private final AccumulatorFactoryBinder factory;
 
     public InternalAggregationFunction(String name, List<Type> parameterTypes, Type intermediateType, Type finalType, boolean decomposable, boolean orderSensitive, AccumulatorFactoryBinder factory)
```

Comment:
Drop this overload in a separate commit at the end of this PR. You may defer this until when you are ready to merge, but don't forget.